File: app/src/main/java/com/callmangement/ui/iris_derivery_installation/adapter/IrisInstalledPendingAdapter.java
package com.callmangement.ui.iris_derivery_installation.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.callmangement.databinding.ItemIrisInstallationPendingBinding;
import com.callmangement.ui.iris_derivery_installation.Model.IrisInstallationPendingListResp;

import java.util.ArrayList;

public class IrisInstalledPendingAdapter extends RecyclerView.Adapter<IrisInstalledPendingAdapter.MyViewHolder> {
    private final ArrayList<IrisInstallationPendingListResp.Datum> irisInsDataArrayList;
    private final Context context;

    public IrisInstalledPendingAdapter(ArrayList<IrisInstallationPendingListResp.Datum> irisInsDataArrayList, Context context) {
        this.irisInsDataArrayList = irisInsDataArrayList;
        this.context = context;
    }

    // data is passed into the constructor
    // inflates the row layout from xml when needed
    @NonNull
    @Override
    public IrisInstalledPendingAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        ItemIrisInstallationPendingBinding itemIrisInstallationPendingBinding = ItemIrisInstallationPendingBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new MyViewHolder(itemIrisInstallationPendingBinding);
    }

    // binds the data to the TextView in each row
    @Override
    public void onBindViewHolder(IrisInstalledPendingAdapter.MyViewHolder holder, int position) {

        IrisInstallationPendingListResp.Datum model = irisInsDataArrayList.get(position);

        holder.binding.tvDealername.setText(model.getDealerName());
        holder.binding.tvFpscode.setText(model.getFpscode());
        holder.binding.tvDealermobno.setText(model.getDealerMobileNo());
        holder.binding.tvTicketno.setText(model.getTicketNo());
        holder.binding.txtDistrict.setText(model.getDistrictName());
        holder.binding.txtBlockName.setText(model.getBlockName());

    }

    // total number of rows
    @Override
    public int getItemCount() {
        return irisInsDataArrayList.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        private final ItemIrisInstallationPendingBinding binding;

        public MyViewHolder(ItemIrisInstallationPendingBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
            binding.getRoot().setOnClickListener(v -> {
            });
        }

        @Override
        public void onClick(View v) {
        }
    }
}

File: app/src/main/java/com/callmangement/ui/iris_derivery_installation/adapter/IrisInstalledPendingAdapter.kt
package com.callmangement.ui.iris_derivery_installation.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.callmangement.databinding.ItemIrisInstallationPendingBinding
import com.callmangement.ui.iris_derivery_installation.Model.IrisInstallationPendingListResp

class IrisInstalledPendingAdapter(
    private val irisInsDataArrayList: ArrayList<IrisInstallationPendingListResp.Datum>,
    private val context: Context
) :
    RecyclerView.Adapter<IrisInstalledPendingAdapter.MyViewHolder>() {
    // data is passed into the constructor
    // inflates the row layout from xml when needed
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemIrisInstallationPendingBinding = ItemIrisInstallationPendingBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return MyViewHolder(itemIrisInstallationPendingBinding)
    }

    // binds the data to the TextView in each row
    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val model = irisInsDataArrayList[position]

        holder.binding.tvDealername.text = model.dealerName
        holder.binding.tvFpscode.text = model.fpscode
        holder.binding.tvDealermobno.text = model.dealerMobileNo
        holder.binding.tvTicketno.text = model.ticketNo
        holder.binding.txtDistrict.text = model.districtName
        holder.binding.txtBlockName.text = model.blockName
    }

    // total number of rows
    override fun getItemCount(): Int {
        return irisInsDataArrayList.size
    }

    class MyViewHolder(val binding: ItemIrisInstallationPendingBinding) :
        RecyclerView.ViewHolder(binding.root), View.OnClickListener {
        init {
            binding.root.setOnClickListener { v: View? -> }
        }

        override fun onClick(v: View) {
        }
    }
}


